api2-war/src/main/java/eu/europeana/api2/v2/utils/ApiKeyUtils.java

```diff
@@ -66,21 +77,90 @@ public LimitResponse checkLimit(String wskey, String url, RecordType recordType,
             LOG.debug("Setting default request number; (checklimit disabled): {} ", requestNumber);
 
         } catch (DatabaseException e) {
-            LOG.error("Error retrieving apikey", e);
+            LOG.error(ERRORRETRIEVING, e);
             ApiKeyException ex = new ApiKeyException(ProblemType.APIKEY_INVALID, wskey);
             ex.initCause(e);
             throw ex;
         } catch (org.hibernate.exception.JDBCConnectionException |
                  org.springframework.transaction.CannotCreateTransactionException ex) {
             // EA-1537 we sometimes have connection problems with the database, so we simply log and do not validate
             // keys when that happens
-            LOG.error("Unable to validate apikey", ex);
+            LOG.error(UNABLETOVALIDATE, ex);
+            requestNumber = 998;
+            apiKey = createTempApiKey(wskey);
+        }
+        return new LimitResponse(apiKey, requestNumber);
+    }
+
+    /**
+     * Validates the supplied apikey using the apikey service
+     * <p>
+     * NOTE: for the time being, this method mimics the user response of the old checkLimit() method with regards to
+     * the JSON format of the output, the returned requestnumber and messages and format of possible error conditions,
+     * including providing a temporary validated key if the apikey service cannot be reached
+     *
+     * @param key The user's public apikey
+     * @return A {@link LimitResponse} consisting of the apiKey and current request number
+     * @throws ApiKeyException {@link ApiKeyException} if an unregistered or unauthorised key is provided
+     *
+     */
+    public LimitResponse validateApiKey(String key) throws ApiKeyException {
+        if (StringUtils.isBlank(key)) {
+            throw new ApiKeyException(ProblemType.APIKEY_MISSING, null, HttpStatus.SC_BAD_REQUEST);
+        }
+
+        ApiKey apiKey;
+        long requestNumber;
+
+        try (CloseableHttpClient client = HttpClients.createDefault()) {
```

Comment:
IIIRC by default http client will only allow 5 connections to 1 single host. This led to a bottleneck in IIIF Manifest, so it could also be problematic here. I think we should raise the number of connections to be safe. See ManifestService for code how to do that.

Reply:
I did, and also moved the initialising code to the class constructor (which I should have done right away).